include/ctype.h

```diff
@@ -397,6 +397,23 @@ int tolower(int c);
 int tolower_l(int c, locale_t locale);
 #endif
 
+#if defined(__LIBCXXTOOLCHAIN__) && defined(__GNUC__)
```

Comment:
why not directly define these macros

Reply:
I don't understand your point? __GNUC__ is defined by GCC to detect that this the compiler in use and __LIBCXXTOOLCHAIN__ is defined by ourself through the Makefile?

Reply:
These functions are always provided by glibc, and are called by many libraries, not only libcxx, so it's better to alwasy implement them without checking some precondition.

Reply:
Oh OK I get your point. I wasn't sure about other compilers (since the flag is C++ toolchain and not specifically GCC C++) so I focused on filtering in case of collision with other compilers. If you think that it's OK without precondition the patch will be way simpler :)